src/vs/workbench/contrib/notebook/browser/view/cellParts/codeCell.ts

```diff
@@ -50,13 +50,52 @@ export class CodeCell extends Disposable {
 
 		const editorHeight = this.calculateInitEditorHeight();
 		this.initializeEditor(editorHeight);
-		this.registerEditorOptionsListener();
-		this.registerViewCellStateChange();
-		this.registerFocusModeTracker();
-		this.registerEditorLayoutListeners();
+
+		this.registerViewCellLayoutChange();
+		this.registerCellEditorEventListeners();
 		this.registerDecorations();
 		this.registerMouseListener();
 
+		this._register(this.viewCell.onDidChangeState(e => {
+			this.cellParts.forEach(cellPart => {
+				cellPart.updateState(this.viewCell, e);
+			});
+
+			if (e.outputIsHoveredChanged) {
+				this.updateForOutputHover();
+			}
+
+			if (e.inputCollapsedChanged) {
+				this.templateData.progressBar.updateForCellState(this.viewCell);
```

Comment:
The progress bar can be a cell part right?

Reply:
@roblourens yes, it would make good sense to make it a cell part.